### PR TITLE
[storage] Move CountryTree realization to cpp.

### PR DESCRIPTION
**Pull request by tatiana-yan**

### CODE REVIEW

**Comment by tatiana-yan** [comment]
JTMAC

File: storage/country_tree.hpp [hunk 1]
```diff
@@ -5,7 +5,6 @@
 
 #include "base/assert.hpp"
```
Comment (mpimenov) [inline storage/country_tree.hpp]: move to cpp

File: storage/country_tree.hpp [hunk 2]
```diff
@@ -5,7 +5,6 @@
...
 
-#include <algorithm>
```
Comment (mpimenov) [inline storage/country_tree.hpp]: include string

File: storage/country_tree.hpp [hunk 3]
```diff
@@ -152,98 +88,28 @@ class CountryTree
...
-    {
-      if (node->ChildrenCount() == 0)
-        return node;
-    }
-    return nullptr;
-  }
+  Node const * const FindFirstLeaf(CountryId const & key) const;
 
 private:
   using CountryTreeHashTable = std::multimap<CountryId, Node *>;
```
Comment (mpimenov) [inline storage/country_tree.hpp]: s/HashTable/Map/g

and please drop this alias

File: storage/country_tree.cpp [hunk 4]
```diff
@@ -10,6 +10,7 @@
 
 #include "3party/jansson/myjansson.hpp"
```
Comment (mpimenov) [inline storage/country_tree.cpp]: 3party after std

Comment (tatiana-yan) [reply storage/country_tree.cpp]: done

File: storage/country_tree.hpp (hunk 3 above)
Comment (tatiana-yan) [reply storage/country_tree.hpp]: done

File: storage/country_tree.hpp (hunk 2 above)
Comment (tatiana-yan) [reply storage/country_tree.hpp]: done

File: storage/country_tree.hpp (hunk 1 above)
Comment (tatiana-yan) [reply storage/country_tree.hpp]: done

**Pull request merged**